CorsixTH/Lua/dialogs/edit_room.lua

```diff
@@ -1202,49 +1249,61 @@ function UIEditRoom:setDoorBlueprint(orig_x, orig_y, orig_wall)
     flags = 0
     y2 = y2 - 1
   end
-
-  self.blueprint_door.valid = checkDoorWalls(x, y, wall, self.room_type.swing_doors)
-  if self.blueprint_door.valid then
-    -- Ensure that the door isn't being built on top of an object
-    local flag_names
-    if wall == "west" then
-      flag_names = {"buildableNorth", "buildableSouth"}
-    else
-      flag_names = {"buildableWest", "buildableEast"}
-    end
-    local player_id = self.ui.hospital:getPlayerIndex()
-
-    if not validDoorTile(x, y, player_id, flag_names) or
-        not validDoorTile(x2, y2, player_id, flag_names) then
+  local world = self.ui.app.world
+  -- invalid_tile used to select the individual blueprint that is blocked
+  local invalid_tile = checkDoorWalls(x, y, wall, self.room_type.swing_doors)
+  self.blueprint_door.valid = invalid_tile == 0
+  -- Ensure that the door isn't being built on top of an object
+  local player_id = self.ui.hospital:getPlayerIndex()
+  if not validDoorTile(x, y, player_id, world) or
+      not validDoorTile(x2, y2, player_id, world) then
+    self.blueprint_door.valid = false
+    invalid_tile = 4
```

Comment:
Shouldn't this be ```+4```

Reply:
I have another push for comment, as this actually needed to be a bitwise operation and due to lack of 5.1 support for bitwise operations I've got a bit of a workaround in place to deal with that.